lib/Controller/Report.php

```diff
@@ -1090,6 +1106,68 @@ public function savedReportExport(Request $request, Response $response, $id, $na
         ));
     }
 
+    /**
+     * Saved Report Convert Form
+     * @param Request $request
+     * @param Response $response
+     * @param $id
+     * @return \Psr\Http\Message\ResponseInterface|Response
+     * @throws AccessDeniedException
+     * @throws GeneralException
+     * @throws NotFoundException
+     * @throws \Xibo\Support\Exception\ControllerNotImplemented
+     */
+    public function convertSavedReportForm(Request $request, Response $response, $id)
+    {
+        $savedReport = $this->savedReportFactory->getById($id);
+
+        if (!$this->getUser()->checkDeleteable($savedReport)) {
+            throw new AccessDeniedException(__('You do not have permissions to convert this saved report'));
+        }
+
+        $data = [
+            'savedReport' => $savedReport
+        ];
+
+        $this->getState()->template = 'savedreport-form-convert';
+        $this->getState()->setData($data);
+
+        return $this->render($request, $response);
+    }
+
+    /**
+     * Converts a Saved Report from Schema Version 1 to 2
+     * @param Request $request
+     * @param Response $response
+     * @param $id
+     * @param $name
+     * @return \Psr\Http\Message\ResponseInterface|Response
+     * @throws GeneralException
+     * @throws \Xibo\Support\Exception\ControllerNotImplemented
+     */
+    public function savedReportConvert(Request $request, Response $response, $id, $name)
+    {
+        $savedReport = $this->savedReportFactory->getById($id);
+
+        if ($savedReport->schemaVersion == 2) {
+            throw new AccessDeniedException(__('You cannot convert this saved report'));
```

Comment:
Access denied doesn't seem right? Perhaps a "General Exception" and the message "This report has already been converted to the latest version."